omod-1.9/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_9/EncounterController1_9Test.java

```diff
@@ -500,6 +501,53 @@ public void shouldUnVoidAnEncounter() throws Exception {
 		
 	}
 	
+	@Test
+	public void shouldFetchAllPatientEncounterIncludingVoidedPatientEncountersIfIncludeAllParamIsSetToTrue()
+	        throws Exception {
+		String encounterUuuid = "c59b3942-4fdd-11e5-8c3c-410f8777c163";
+		String patientUuid = "da7f524f-27ce-4bb2-86d6-6d1d05312bd5";
+		Patient patient = Context.getPatientService().getPatientByUuid(patientUuid);
+		
+		Encounter enc = Context.getEncounterService().getEncounterByUuid(encounterUuuid);
+		Context.getEncounterService().voidEncounter(enc, "bcbcb");
+		Context.getEncounterService().saveEncounter(enc);
+		
+		//list of non voided encounters
+		List<Encounter> encs = Context.getEncounterService().getEncountersByPatient(patient.getGivenName());
+		Assert.assertEquals(1, encs.size());
+		
+		MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+		req.addParameter("patient", patientUuid);
+		req.addParameter("includeAll", "true");
+		
+		SimpleObject result = deserialize(handle(req));
+		List<Object> encounters = result.get("results");
+		Assert.assertEquals(2, encounters.size());
+	}
+	
+	@Test
+	public void shouldFetchOnlyNonVoidedPatientEncountersIfIncludeAllParamIsSetToFalse() throws Exception {
+		String encounterUuuid = "c59b3942-4fdd-11e5-8c3c-410f8777c163";
```

Comment:
Does this test fail without your changes?

Reply:
well thats was the default situation even before my changes

Reply:
Then what is the purpose of the test? Did you read about a unit test failing first and then make changes to make it pass? TDD

Reply:
removed the unnecesary test unit